galaxy_ng/app/api/ui_v2/serializers.py

```diff
@@ -41,16 +43,17 @@ def get_groups(self, obj):
         return groups_serializer.data
 
     def get_teams(self, obj):
-        teams = Team.objects.filter(users=obj)
+        """Return all 'local' team member assignments."""
+        roledef = RoleDefinition.objects.get(name='Galaxy Team Member')
+        assignments = RoleUserAssignment.objects.filter(
+            user=obj, role_definition=roledef
+        ).values_list('object_id', flat=True)
+        teams = Team.objects.filter(pk__in=list(assignments))
```

Comment:
That will only return teams that user has "local" membership to. The intended approach is to treat membership as a user permission, so `Team.access_qs(obj, 'member')` should get the desired queryset, if including both local & JWT role membership.

But I'd also prefer to not specify this as a part of local serializers. Because that enforces tighter coupling between the resources and the permission system. We want to go in the direction of separated, and eventually external, permissions management.

Reply:
How & when does a user get a non-local member role assignment at this point? The jwt won't do it, the tests don't do it ... what is supposed to? The sysadmin directly via the API? What would make them want to do that? What necessitates "Team Member" vs "Galaxy Team Member"? Does it this only make sense when reverse sync is a thing?

Reply:
>How & when does a user get a non-local member role assignment at this point? The jwt won't do it, the tests don't do it ... what is supposed to?

The JWT is supposed to do it. The "Team Member" and other roles are owned by the JWT auth. If someone is said to be a member of a team by the JWT auth, the JWT auth class should assign the "Team Member" role. This gives consistency across other services, which have a role definition by that same name.

Thus, the JWT roles can be called shared roles. Those role assignments are served by a common provider (which is also the resource provider) to multiple services. So assigning a user to that role in the resource provider effectively gives them that permission to all the services. Some _other_ permissions are only stored in the local API, which is a source of confusion, but the UI kind of has its approach to smooth this out.